src/sparse/impl/KokkosSparse_spmv_struct_impl.hpp

```diff
@@ -684,38 +719,52 @@ spmv_struct_beta_no_transpose (const int stencil_type,
     return;
   }
 
-  int team_size = -1;
   int vector_length = -1;
   int nnzPerRow = -1;
-  int64_t rows_per_thread = -1;
+  int team_size_int = -1;
+  int team_size_ext = -1;
+  int64_t rows_per_thread_int = -1;
+  int64_t rows_per_thread_ext = -1;
   int64_t numInteriorPts = 0;
+  int64_t numExteriorPts = 0;
 
   if(structure.extent(0) == 1) {
     numInteriorPts = structure(0) - 2;
+    numExteriorPts = 2;
     vector_length = 1;
   } else if(structure.extent(0) == 2) {
     numInteriorPts = (structure(1) - 2)*(structure(0) - 2);
+    numExteriorPts = 2*(structure(1) + structure(0) - 2);
     if(stencil_type == 1) {
       vector_length = 2;
     } else if(stencil_type == 2) {
-      vector_length = 4;
+      vector_length = 2;
     }
   } else if(structure.extent(0) == 3) {
     numInteriorPts = (structure(2) - 2)*(structure(1) - 2)*(structure(0) - 2);
+    numExteriorPts = structure(2)*structure(1)*structure(0) - numInteriorPts;
     if(stencil_type == 1) {
       vector_length = 2;
     } else if(stencil_type == 2) {
       vector_length = 8;
     }
   }
 
-  int64_t rows_per_team = spmv_struct_launch_parameters<execution_space>(numInteriorPts,
-                                                                         A.nnz(),
-                                                                         nnzPerRow,
-                                                                         rows_per_thread,
-                                                                         team_size,
-                                                                         vector_length);
-  int64_t worksets = (numInteriorPts + rows_per_team - 1) / rows_per_team;
+  int64_t rows_per_team_int = spmv_struct_launch_parameters<execution_space>(numInteriorPts,
```

Comment:
Looks like some odd tabs are left

Reply:
I ran `untabify` on the file so this should have been taken care of, unless you want the alignment of function parameters to be handled differently?

Reply:
As long as it doesn't have tabs, and easily readable I am ok.